src/main/java/com/google/sps/servlets/Util.java

```diff
@@ -0,0 +1,145 @@
+  private static final String ALL_DEVICES_ENDPOINT = "https://www.googleapis.com/admin/directory/v1/customer/my_customer/devices/chromeos";
+  private static final String DEFAULT_MAX_DEVICES = "200"; //is limited to effectively 200
+  private static final String DEFAULT_SORT_ORDER = "ASCENDING";
+  private static final String DEFAULT_PROJECTION = "FULL";
+  private static final DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
+  private static final String API_KEY = getAPIKey(); 
+
+  public static List<ChromeOSDevice> getAllDevices(String userId) throws IOException {
+    final String refreshToken = getRefreshToken(userId);
+    File file = new File(Util.class.getResource(CLIENT_SECRET_FILE).getFile());
+    final GoogleClientSecrets clientSecrets =
+    GoogleClientSecrets.load(
+      JacksonFactory.getDefaultInstance(), new FileReader(file));
+    final String clientId = clientSecrets.getDetails().getClientId();
+    final String clientSecret = clientSecrets.getDetails().getClientSecret();
+    final String accessToken = getAccessToken(refreshToken, clientId, clientSecret);
+    ListDeviceResponse resp = getDevicesResponse(EMPTY_PAGE_TOKEN, accessToken);
+    final List<ChromeOSDevice> allDevices = new ArrayList<>(resp.getDevices());
+    while (resp.hasNextPageToken()) {
+      final String pageToken = (String) resp.getNextPageToken();
+      resp = getDevicesResponse(pageToken, accessToken);
+      allDevices.addAll(resp.getDevices());
+    }
+    return allDevices;
+  }
+
+  public static String getAPIKey() {
+    try {
+      File file = new File(Util.class.getResource(API_KEY_FILE).getFile());
+      String str = FileUtils.readFileToString(file);
+      return str;
+    } catch (IOException e) {
+      System.out.println(e.getMessage());
+    }
+    return EMPTY_API_KEY;
+  }
+
+  private static String getRefreshToken(String userId) throws IOException {
+    Query query = new Query("RefreshToken").setFilter(FilterOperator.EQUAL.of("userId", userId));
+    PreparedQuery results = datastore.prepare(query);
+    System.out.println(results.countEntities());
+    try {
+      Entity entity = results.asSingleEntity();
+      String refreshToken = (String) entity.getProperty("refreshToken");
+      return refreshToken;
+    } catch (PreparedQuery.TooManyResultsException e) {
+      System.out.println(e);
+      throw new IOException("Error while getting refresh token");
+    }
+  }
+
+  private static String getAccessToken(String refreshToken, String clientId, String clientSecret) throws IOException {
+    try {
+      GoogleTokenResponse response =
+      new GoogleRefreshTokenRequest(new NetHttpTransport(), new JacksonFactory(),
+        refreshToken, clientId, clientSecret).execute();
+      return response.getAccessToken();
+    } catch (TokenResponseException e) {
+      if (e.getDetails() != null) {
+        System.out.println(e.getMessage());
+      }
+    }
+    return INVALID_ACCESS_TOKEN;
```

Comment:
It's better to just throw the exception, instead of returning an invalid token and allowing the exception to be thrown at a later stage that you may even not know where it is.